integration_tests/steps/two_sided_pay_as_clear.py

```diff
@@ -105,20 +105,26 @@ def test_offer_bid_market_clearing_rate_files(context):
 
 @then('one-on-one matching of offer & bid in PAC happens at bid rate')
 def one_on_one_matching_at_clearing_rate_at_bid_rate(context):
-    assert all(isclose(clearing, 30.0)
-               for clearing in
-               bid_offer_matcher.matcher.match_algorithm.state.clearing.values())
+    count = 0
+    for clearing in bid_offer_matcher.matcher.match_algorithm.state.clearing.values():
+        count += 1
+        assert all(isclose(cv[0], 30.0) for cv in clearing.values())
+    assert count > 0
 
 
 @then('clearing rate is the bid rate of last matched bid')
 def clearing_rate_at_last_matched_bid_rate(context):
-    assert all(isclose(clearing, 15.0)
-               for clearing in
-               bid_offer_matcher.matcher.match_algorithm.state.clearing.values())
+    count = 0
+    for clearing in bid_offer_matcher.matcher.match_algorithm.state.clearing.values():
+        count += 1
+        assert all(isclose(cv[0], 15.0) for cv in clearing.values())
+    assert count > 0
 
 
 @then('clearing rate is equal to the bid_rate')
 def clearing_rate_at_bid_rate(context):
-    assert all(isclose(clearing, 25.0)
-               for clearing in
-               bid_offer_matcher.matcher.match_algorithm.state.clearing.values())
+    count = 0
+    for clearing in bid_offer_matcher.matcher.match_algorithm.state.clearing.values():
+        count += 1
+        assert all(isclose(cv[0], 25.0) for cv in clearing.values())
+    assert count > 0
```

Comment:
Hmm, AFAIU, `bid_offer_matcher.matcher.match_algorithm.state.clearing` is a `Dict[DateTime, Tuple]`.
And you want to check if the clearing is at 25 for all market slots here right?
So I suggest a slightly modified version of the original code:
```suggestion
     assert all(isclose(cv[0], 25.0) for cv in bid_offer_matcher.matcher.match_algorithm.state.clearing.values())
```

Reply:
`bid_offer_matcher.matcher.match_algorithm.state.clearing` is a `Dict[market_id, Dict[DateTime, Clearing]]`

Reply:
Hmm, ok. This makes me think. Why do we need to save both market_id and time_slot in the structure? Isn't one of them sufficient? 
This is related to my comment [here](https://github.com/gridsingularity/d3a-interface/pull/246/files#r705950487)

Reply:
There are separate plots for each markets, and each markets can have multiple clearings. Hope this answers your question.